lib/node-token-validation/src/config/TokenValidationParameters.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Configuration } from "./Configuration";
+import { JwtConstants } from "../utils/Constants";
+import { KeyLike } from "jose";
+
+export type TokenValidationParameters = {
+    // Valid claims
+    validAlgorithms: Array<string>,
+    validIssuers: Array<string>,
+    validAudiences?: Array<string>,
+    validTypes?: Array<string>,
+    // Signature validation
+    issuerSigningKeys?: Array<KeyLike|Uint8Array>,
+    issuerSigningJwksUri?: string,
+    // Other
+    requireExpirationTime?: Boolean,
+    requireSignedTokens?: Boolean,
+    subject?: string
+    nonce?: string,
+    code?: string,
+    accessTokenForAtHash?: string,
+};
+
+export type ValidationParameters = {
+    // Valid claims
+    validAlgorithms: Required<Array<string>>,
+    validIssuers: Array<string>,
+    validAudiences: Required<Array<string>>,
+    validTypes?: Array<string>,
+    // Signature validation
+    issuerSigningKeys?: Array<KeyLike|Uint8Array>,
+    issuerSigningJwksUri?: string,
+    // Other
+    requireExpirationTime: Required<Boolean>,
+    requireSignedTokens: Required<Boolean>,
+    subject?: string
+    nonce?: string,
+    code?: string,
+    accessTokenForAtHash?: string,
+};
+
+export function buildTokenValidationParameters(params: TokenValidationParameters, config: Configuration): ValidationParameters {
+    const DEFAULT_VALIDATION_PARAMS = { // Are there defaults for these? Which ones should not have defaults?
+        validAlgorithms: [JwtConstants.RSA_256],
+        validIssuers: [],
+        validAudiences: [`api://${config.auth.clientId}`, config.auth.clientId], // Is this still needed now in this new library?
```

Comment:
One other nuance is to accont for trailing slash on the `api://{client-id}/` format.

Reply:
I'm curious what other libraries do here...they may require an application to provide these explicitly.

Reply:
After some investigation, it seems that other libraries require providing these explicitly, which seems better. Removed these defaults.